### DISCUSSION TITLE
Adding sw8 on the browser side cannot directly connect with the java side

### DISCUSSION THREAD

**Discussion by Minya01** [apache/skywalking#10203] [Q&A]
Hello, we initialized skywalking (“Skywalking-client-js”:”^ 0.9.0”) on the browser side, and initiated the interface request ( Interface A) through the SW8 protocol. We found the following three situations:

> ① Interface A is forwarded by NodeJs (“Skywalking-backend-js”:”^ 0.5.1”) and can be concatenated with the server by traceId on the browser side.

<img width="937" alt="image" src="https://user-images.githubusercontent.com/19857248/209496286-3f61231d-a797-442c-ab3d-76e0f2625ffe.png">

> ② Interface A is forwarded through Nginx and can be concatenated with the server through traceId on the browser side.

<img width="810" alt="image" src="https://user-images.githubusercontent.com/19857248/209496982-42c61b0f-4f76-4a78-99dd-8fd0443dbd80.png">

> ③ The interface directly requests the Java side, and can not be concatenated through the browser-side traceid, so the Java service regenerates a traceid by itself.

<img width="1774" alt="image" src="https://user-images.githubusercontent.com/19857248/209496920-3fbb1f41-9414-4989-b18b-459d24ac3cb2.png">

Excuse me, is the third case a bug or do we need to do something about it?

**Comment by wu-sheng** [comment]
I believe the reality is not as same as your description. Because the agent doesn't know the existence of the browser agent. It is just identifying whether `sw8` header exists. 

Ref prototocl, https://skywalking.apache.org/docs/main/next/en/api/x-process-propagation-headers-v3/

**Reply by Minya01** [reply]
> I believe the reality is not as same as your description. Because the agent doesn't know the existence of the browser agent. It is just identifying whether `sw8` header exists.
> 
> 我相信现实和你描述的不一样。因为代理不知道浏览器代理的存在。它只是确定 sw8报头是否存在。
> 
> Ref prototocl, https://skywalking.apache.org/docs/main/next/en/api/x-process-propagation-headers-v3/
> 
> 裁判 https://skywalking.apache.org/docs/main/next/en/api/x-process-propagation-headers-v3/

We used skywalking-client-js, with SW8 headers, and forwarding via nodejs or Nginx is fine, except when directly requesting Java (springMVC) .
<img width="1198" alt="image" src="https://user-images.githubusercontent.com/19857248/209498409-2b6093d9-478e-4418-a465-fc8faa2bd5ff.png">

In the problem, figure 1 is OK to forward through nodeJs, figure 2 is OK to Nginx, and Figure 3 calls directly to the back-end interface and finds that the back-end does not use the browser-side TraceId.

What could have caused this?

**Reply by wu-sheng** [reply]
I don't think this has anything about the codes. This is your env setting up making this. Some firewalls or network policies removed unexpected headers(such as security policy) when you change the request path.
`Brower -> Nginx -> Java server` is as same as `Browser -> Java server` from the Java agent perspective.

**Reply by guuuuuuuuuuuu** [reply]
Hello, let me describe the actual situation.
We use the postman test to add a sw8 generated by the js client to the request header.
The request is connected to the node service of skywalking. It is found that the request link is connected.
The request is connected to the Apisix of skywalking. It is found that the request link is also connected.
However, the request accesses the skywalking java service, but finds that the java service is the first node, and the previous link is missing.

Both js client and java client are the latest versions.

We don't know how to troubleshoot. Please give me some guidance. Thank you！

**Reply by Superskyyy** [reply]
> Hello, let me describe the actual situation. We use the postman test to add a sw8 generated by the js client to the request header. The request is connected to the node service of skywalking. It is found that the request link is connected. The request is connected to the Apisix of skywalking. It is found that the request link is also connected. However, the request accesses the skywalking java service, but finds that the java service is the first node, and the previous link is missing.
> 
> Both js client and java client are the latest versions.
> 
> We don't know how to troubleshoot. Please give me some guidance. Thank you！

Have you double checked the received header on the java service side? Essentially making sure the sw8 header is still there to be identified by the Java agent.
